src/test/java/uk/gov/companieshouse/api/testdata/service/impl/AcspProfileServiceImplTest.java

```diff
@@ -83,6 +92,142 @@ void createAcspProfileWithDefaultValues() throws DataException {
         assertEquals(0L, captured.getVersion());
     }
 
+    @Test
+    void createAcspProfileWithAmlDetails() throws DataException {
+        AcspProfileSpec spec = getAcspProfileSpec();
+
+        when(randomService.getString(8)).thenReturn("randomId");
+        AcspProfile savedProfile = getAcspProfile(spec);
+
+        when(repository.save(any(AcspProfile.class))).thenReturn(savedProfile);
+
+        AcspProfileData result = service.create(spec);
+
+        assertNotNull(result);
+        assertEquals(savedProfile.getAcspNumber(), result.getAcspNumber());
+
+        ArgumentCaptor<AcspProfile> captor = ArgumentCaptor.forClass(AcspProfile.class);
+        verify(repository).save(captor.capture());
+
+        AcspProfile captured = captor.getValue();
+        assertEquals("randomId", captured.getId());
+        assertEquals("randomId", captured.getAcspNumber());
+        assertEquals(spec.getStatus(), captured.getStatus());
+        assertEquals(spec.getType(), captured.getType());
+        assertEquals("Test Data Generator randomId Company Ltd", captured.getName());
+        assertEquals("/authorised-corporate-service-providers/randomId", captured.getLinksSelf());
+        assertEquals(0L, captured.getVersion());
+
+        assertNotNull(captured.getAmlDetails());
+        assertEquals(3, captured.getAmlDetails().size());
+        assertEquals("Supervisory Body 1",
+                captured.getAmlDetails().get(0).getSupervisoryBody());
+        assertEquals("Membership Details 1",
+                captured.getAmlDetails().get(0).getMembershipDetails());
+        assertEquals("Supervisory Body 2",
+                captured.getAmlDetails().get(1).getSupervisoryBody());
+        assertEquals("Membership Details 2",
+                captured.getAmlDetails().get(1).getMembershipDetails());
+        assertEquals("Supervisory Body 3",
+                captured.getAmlDetails().get(2).getSupervisoryBody());
+        assertEquals("Membership Details 3",
+                captured.getAmlDetails().get(2).getMembershipDetails());
+    }
+
+    private AcspProfile getAcspProfile(AcspProfileSpec spec) {
+        AcspProfile savedProfile = createSavedProfile();
+
+        List<AmlDetails> amlDetailsList = new ArrayList<>();
+        for (AmlSpec amlSpec : spec.getAmlDetails()) {
+            AmlDetails amlDetails = new AmlDetails();
+            amlDetails.setSupervisoryBody(amlSpec.getSupervisoryBody());
+            amlDetails.setMembershipDetails(amlSpec.getMembershipDetails());
+            amlDetailsList.add(amlDetails);
+        }
+        savedProfile.setAmlDetails(amlDetailsList);
+        return savedProfile;
+    }
+
+    private AcspProfileSpec getAcspProfileSpec() {
```

Comment:
This seems to be creating a spec for active ltd and 3 aml memberships and it is used only in one test so not sure if it is worth having it in a separate method. If it was in the test we could refer to the `amlSpec1`, `amlSpec2` and `amlSpec3` object for the assertions rather than hardcode the sup body names and memberships?